### PR TITLE
feat: adding s3 & cloudfront parsers

### PR DESCRIPTION
**Pull request by GuyMoses**

### CODE REVIEW

File: src/parsers/eventParser.js [hunk 1]
```diff
@@ -135,6 +175,55 @@ export const parseSqsEvent = event => {
   return new_sqs_event;
 };
 
+export const parseS3Event = event => {
+  const new_s3_event = {};
```
Comment (GuyMoses) [inline src/parsers/eventParser.js:179]: I don't know why, but in this page every parser is written in snake-case and not camel-case

**Comment by efimk-lu** [comment]
@doriaviram & @GuyMoses why not adding ts support for this file? I added it as a request for this ticket. Feasible or not?

**Review by uri-p — APPROVED** [review]

File: src/parsers/eventParser.test.js [hunk 2]
```diff
@@ -402,4 +402,142 @@ describe('event parser', () => {
...
+            configurationId: 'a078ca2d-53a8-45d4-a621-260a439876d8',
+            bucket: {
+              name: 'testingbuckets3testing',
+              ownerIdentity: { principalId: 'A2QVTU9T5VMOU3' },
+              arn: 'arn:aws:s3:::testingbuckets3testing',
+            },
+            object: {
+              key: 'Screen+Shot+2020-05-27+at+12.37.36.png',
+              size: 61211,
+              eTag: '714ee5196a5c0a6e6b9019caa7b6e970',
```
Comment (uri-p) [inline src/parsers/eventParser.test.js:433]: Should not exists no?

File: src/parsers/eventParser.test.js [hunk 3]
```diff
@@ -402,4 +402,142 @@ describe('event parser', () => {
...
+            'x-amz-request-id': '318F33BA8C4CBDC5',
+            'x-amz-id-2':
+              'VyRyYV/2vjikRUkRoH2WPH6M5WcAjNSGXG8Qtd1uEfbklnTusaDEz/jQPdLQgf2tZLjRuq4MgZFcVFpQJgZLJfiGUoh7IBhU',
+          },
+          s3: {
+            s3SchemaVersion: '1.0',
+            configurationId: 'a078ca2d-53a8-45d4-a621-260a439876d8',
+            bucket: {
+              name: 'testingbuckets3testing',
+              ownerIdentity: { principalId: 'A2QVTU9T5VMOU3' },
```
Comment (uri-p) [inline src/parsers/eventParser.test.js:427]: Should not exists no?

File: src/parsers/eventParser.test.js (hunk 2 above)
Comment (uri-p) [inline src/parsers/eventParser.test.js:433]: Should not exists no?

**Review by uri-p — CHANGES_REQUESTED** [review]

File: src/parsers/eventParser.test.js (hunk 2 above)
Comment (GuyMoses) [reply src/parsers/eventParser.test.js:433]: it doesn't exist after parsing

Comment (GuyMoses) [reply src/parsers/eventParser.test.js:433]: it doesn't exist after parsing

File: src/parsers/eventParser.test.js (hunk 3 above)
Comment (GuyMoses) [reply src/parsers/eventParser.test.js:427]: it doesn't exist after parsing

**Review by uri-p — APPROVED** [review]

File: src/parsers/eventParser.js [hunk 4]
```diff
@@ -66,73 +88,149 @@ export const isSqsEvent = event => {
...
+  newCloudfrontEvent['Records'] = [];
+  // Add order keys
+  for (const rec of event['Records']) {
+    const cfRecord = rec['cf'] || {};
+    const newCloudfrontRecordEvent = {};
+    for (const key of CLOUDFRONT_KEYS_ORDER) {
+      if (cfRecord.hasOwnProperty(key) != null) {
+        newCloudfrontRecordEvent[key] = cfRecord[key];
+      }
+    }
```
Comment (doriaviram) [inline src/parsers/eventParser.js:222]: You can do util that will help you reuse this?
That gets a JSON and order and return a JSON

**Review by doriaviram — CHANGES_REQUESTED** [review]

**Review by doriaviram — APPROVED** [review]

**Pull request merged**